scylla/src/transport/connection.rs

```diff
@@ -1199,21 +1223,24 @@ impl Connection {
     }
 
     pub(crate) async fn fetch_schema_version(&self) -> Result<Uuid, QueryError> {
-        let (version_id,) = self
+        let (result, paging_state) = self
             .query_single_page(LOCAL_VERSION, PageSize::default())
-            .await?
-            .single_row_typed()
-            .map_err(|err| match err {
-                SingleRowTypedError::RowsExpected(_) => {
-                    QueryError::ProtocolError("Version query returned not rows")
-                }
-                SingleRowTypedError::BadNumberOfRows(_) => {
-                    QueryError::ProtocolError("system.local query returned a wrong number of rows")
-                }
-                SingleRowTypedError::FromRowError(_) => {
-                    QueryError::ProtocolError("Row is not uuid type as it should be")
-                }
-            })?;
+            .await?;
+        if !paging_state.finished() {
+            warn!("Fetching local schema version returned nonfinished paging state");
+        }
```

Comment:
Not strictly related to this PR: I'm wondering if it's ever possible for that to happen. Maybe unpaged / iter should be used here for correctness?

Reply:
I agree, I see no reason why we should use `query_single_page` here.